caravel/views.py

```diff
@@ -2118,30 +2118,39 @@ def queries(self, last_updated_ms):
     def search_queries(self):
         """Search for queries."""
         query = db.session.query(models.Query)
-        userId = request.args.get('userId')
-        databaseId = request.args.get('databaseId')
-        searchText = request.args.get('searchText')
+        search_user_id = request.args.get('user_id')
+        database_id = request.args.get('database_id')
+        search_text = request.args.get('search_text')
         status = request.args.get('status')
+        from_time = request.args.get('from')
+        to_time = request.args.get('to')
 
-        if userId != 'null':
+        if search_user_id:
             # Filter on db Id
-            query = query.filter(models.Query.user_id == userId)
+            query = query.filter(models.Query.user_id == search_user_id)
 
-        if databaseId != 'null':
+        if database_id:
             # Filter on db Id
-            query = query.filter(models.Query.database_id == databaseId)
+            query = query.filter(models.Query.database_id == database_id)
 
-        if status != 'null':
+        if status:
             # Filter on status
             query = query.filter(models.Query.status == status)
 
-        if searchText != 'null':
+        if search_text:
             # Filter on search text
-            query = query.filter(models.Query.sql.like('%{}%'.format(searchText)))
+            query = query \
+                .filter(models.Query.sql.like('%{}%'.format(search_text)))
 
-        sql_queries = query.limit(config.get("QUERY_SEARCH_LIMIT")).all()
+        if from_time:
+            query = query.filter(models.Query.start_time > int(from_time))
+
+        if to_time:
+            query = query.filter(models.Query.start_time < int(to_time))
 
+        sql_queries = query.limit(config.get("QUERY_SEARCH_LIMIT")).all()
```

Comment:
QUERY_SEARCH_LIMIT may be not set or absent in the config. Please handle that case.